### DISCUSSION TITLE
A few questions and thank you!

### DISCUSSION THREAD

**Discussion by ruipgil** [wezterm/wezterm#662] [Q&A] [answered]
Hi Wez,

I'm thinking of switching to wezterm as my daily driver, love the performance, simplicity, and feature set. I can finally see a future without the sluggish performance of tmux!

I've got a couple of questions, I'm not sure if some are bugs or feature requests, so I'm reaching out through discussions 🙂 

1. I'm using a Portuguese mac keyboard layout, and can't get { or } to show up. I've got to press alt+shift+8 and alt+shift+9 for those.. it looks like it's not recognizing all the alt. I've tried a combination of settings described here: https://wezfurlong.org/wezterm/config/keys.html#alt--option-key-behavior--composed-keys but I've had no success
2. It doesn't seem to be possible to use space as a leader key. Is that intended? I get this error:
```
while processing a struct of type `LeaderKey` with value:
{
    "key": "SPACE",
    "mods": "CTRL",
}
invalid KeyCode string SPACE
```
3. Finally, I'd like to use ctrl+[HJKL] to switch terminal panes and inside vim! I've looked at https://github.com/christoomey/vim-tmux-navigator and the implementation looks straightforward, and it looks like something that can be done with the wezterm config. It goes as follows: check if the current process of the pane is vim or nvim, if so send the keys downward (nop,) otherwise select a pane based on the key pressed. Do you have any pointers?

Again, if you feel like this is not the right place and prefer to have issues, let me know and I'll move this there.

And thank you so much for your work.

**Comment by wez** [comment, marked as answer]
For 1. it sounds like you might be running into https://github.com/wez/wezterm/issues/601#issuecomment-812867374
I don't know what the resolution is for this yet!

For 2., use `key=" "` rather than `key="SPACE"`

For 3, we don't yet have an API that let's you query the process name from your lua config, but you may be able to do it a little indirectly:

* In your base config, define the key assignments for CTRL+[HJKL] as appropriate [ActivatePaneDirection](https://wezfurlong.org/wezterm/config/lua/keyassignment/ActivatePaneDirection.html) actions
* Define an [update-right-status](https://wezfurlong.org/wezterm/config/lua/window-events/update-right-status.html) hook; that hook triggers periodically so that you can set a fancy status area, but it is a good spot to test what is running in a pane
* In that hook, use [pane:get_title()](https://wezfurlong.org/wezterm/config/lua/pane/get_title.html) to examine the title of the pane.  Assuming that vim and nvim both put `vim` in the title, you can check whether the title contains the string `vim`
* If the title contains vim, then use [window:set_config_overrides](https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html) to change the key assignments accordingly.
* otherwise (the title doesn't contain vim), remove the config overrides

**Reply by ruipgil** [reply]
Thank you for the prompt reply. I tried to search for "{" and "}" but that issue didn't show up... I'll try to do a little bit more debugging, maybe even building from source and see what I can find.

For 2, I forgot to mention that I did try that. It doesn't throw an error, but it doesn't work.. I can open an issue for that

For 3, yes, that was my idea. I'll give it a go.

**Reply by ruipgil** [reply]
Also for 3, is it currently possible to pass a function that would dynamically return an action? That should simplify and make the implementation less hacky

**Reply by wez** [reply]
Take a look at the [EmitEvent](https://wezfurlong.org/wezterm/config/lua/keyassignment/EmitEvent.html) action and the [custom events example](https://wezfurlong.org/wezterm/config/lua/wezterm/on.html#custom-events)
